src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkUnrecognizedSectionInspection.kt

```diff
@@ -72,5 +94,13 @@ class SmkUnrecognizedSectionInspection : SnakemakeInspection() {
 
     override fun createOptionsPanel(): JComponent? =
         ListEditForm(SnakemakeBundle.message("INSP.NAME.section.unrecognized.ignored"), ignoredItems).contentPanel
+
+    private fun getSectionBySubsection(name: String) = when (name) {
+        in RULE_OR_CHECKPOINT_SECTION_KEYWORDS -> RULE_KEYWORD
```

Comment:
If I understand your idea correctly, we can't just take section keyword because inspected subsection is already inappropriate for this sectiction and we're using such switch to get the section name, where current subsection can be used